### PR TITLE
Allow separate building of debug and release libraries.

### LINKED ISSUE #184: Split tasks to build debug and release flavors of objc binaries.
we should have j2objcBuildDebug and j2objcBuildRelease.

### PR DESCRIPTION
**Pull request by advayDev1**
fixes #184

### CODE REVIEW

**Comment by advayDev1** [comment]
ptal @confile @brunobowden

**Comment by confile** [comment]
@advayDev1 what means ptal?

**Comment by advayDev1** [comment]
@confile, ptal = please take a (or another) look :)
also i think a FAQ entry might be better than readme.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy [hunk 1]
```diff
@@ -270,12 +270,17 @@ class J2objcNativeCompilation {
...
-            task('buildAllObjcLibraries').configure {
+            task('j2objcBuildObjcDebug').configure {
                 dependsOn binaries.withType(NativeLibraryBinary).matching {
-                    it.buildable
+                    it.buildable && it.buildType.name == 'debug'
+                }
+            }
+            task('j2objcBuildObjcRelease').configure {
+                dependsOn binaries.withType(NativeLibraryBinary).matching {
+                    it.buildable && it.buildType.name == 'release'
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy]: Nice tricks for doing this

**Comment by brunobowden** [comment]
Looks good

GitHub indicates that you have some merge conflicts but I'm sure you'll work it out.

**Comment by confile** [comment]
LGTM

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy (hunk 1 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcNativeCompilation.groovy]: yes, but i think it is unfortunate.  i can't find constants to use for this.  one day they capitalize Debug and the code will stop working =)

**Pull request merged**